test/cmd/router.sh

```diff
@@ -37,6 +37,10 @@ os::cmd::expect_success_and_not_text "oadm router --dry-run --host-network=false
 os::cmd::expect_success_and_not_text 'oadm router --dry-run --host-network=false --host-ports=false -o yaml' 'OPENSHIFT_MASTER'
 os::cmd::expect_success_and_not_text 'oadm router --dry-run --host-network=false --host-ports=false --secrets-as-env -o yaml' 'OPENSHIFT_MASTER'
 os::cmd::expect_success_and_text 'oadm router --dry-run --host-network=false --host-ports=false --secrets-as-env --credentials=${KUBECONFIG} -o yaml' 'OPENSHIFT_MASTER'
+# canonical hostname
+os::cmd::expect_success_and_text 'oadm router --dry-run --router-canonical-hostname=a.b.c.d -o yaml' 'a.b.c.d'
+os::cmd::expect_failure_and_not_text 'oadm router --dry-run --router-canonical-hostname=1a.b.c.d -o yaml' '1a.b.c.d'
```

Comment:
Why is 1a invalid?

https://en.wikipedia.org/wiki/Hostname#Restrictions_on_valid_hostnames